test/update_yml_precomputed_files.py

```diff
@@ -0,0 +1,75 @@
+"""
+Run scipt to update buildstock.csv for tests:
+- yml_precomputed
+- yml_precomputed_outdated
+- yml_precomputed_weight
+"""
+
+import os
+import subprocess
+import yaml
+from pathlib import Path
+import pandas as pd
+
+resstock_dir = Path(__file__).resolve().parents[1]
+project_file = resstock_dir / "project_testing" / "testing_baseline.yml"
+test_dir = resstock_dir / "test" / "tests_yml_files"
+
+
+def load_project_yaml(project_file):
+    with open(project_file) as f:
+        cfg = yaml.load(f, Loader=yaml.SafeLoader)
+    return cfg
+
+
+def save_project_yaml(cfg, project_file):
+    with open(project_file, "w") as f:
+        yaml.dump(cfg, f)
+
+
+def openstudio_exe():
+    return os.environ.get("OPENSTUDIO_EXE", "openstudio")
+
+
+def generate_buildstock():
+    cfg = load_project_yaml(project_file)
+    cfg["sampler"]["args"]["n_datapoints"] = 2
+    cfg["output_directory"] = "test/tests_yml_files/yml_precomputed"
+    tmp_file = project_file.parent / (project_file.stem + "_tmp" + project_file.suffix)
+    save_project_yaml(cfg, tmp_file)
+
+    result = subprocess.run(
+        [openstudio_exe(), "workflow/run_analysis.rb", "-y", tmp_file, "-s"],
```

Comment:
I _think_ you should be able to just use `ruby` (instead of `openstudio`) to run sampling with `run_analysis.rb`. You may need to install additional gems up front. Then you wouldn't need to download and install `openstudio` in this config job.

Reply:
With ruby instead of openstudio.exe, I am getting 
```
workflow/run_analysis.rb:589:in `<main>': uninitialized constant OptionParser (NameError)
```

Reply:
Put a `require 'optparse'` at the top of `run_analysis.rb`.

Reply:
I tried to invoke the script with ruby only on Github action but because run_analysis.rb has a `require_relative '../resources/buildstock'`, which has a `require 'openstudio'`, I can't do it with ruby alone.

Reply:
You should be able to do it with https://github.com/NREL/resstock/pull/1127. Up to you.